docs/dev/table/connectors/kinesis.md

```diff
@@ -0,0 +1,270 @@
+By default, the `AUTO` Credentials Provider is used.
+If the access key ID and secret key are set in the deployment configuration, this results in using the `BASIC` provider.  
+
+A specific Credentials Provider can be **optionally** set using the `properties.aws.credentials.provider` setting.
+Supported values are:
+
+* `AUTO` - Use the default AWS Credentials Provider chain that searches for credentials in the following order: `ENV_VARS`, `SYS_PROPS`, `WEB_IDENTITY_TOKEN`, `PROFILE`, and EC2/ECS credentials provider.
+* `BASIC` - Use access key ID and secret key supplied as configuration. 
+* `ENV_VAR` - Use `AWS_ACCESS_KEY_ID` & `AWS_SECRET_ACCESS_KEY` environment variables.
+* `SYS_PROP` - Use Java system properties `aws.accessKeyId` and `aws.secretKey`.
+* `PROFILE` - Use an AWS credentials profile to create the AWS credentials.
+* `ASSUME_ROLE` - Create AWS credentials by assuming a role. The credentials for assuming the role must be supplied.
+* `WEB_IDENTITY_TOKEN` - Create AWS credentials by assuming a role using Web Identity Token. 
+
+### Start Reading Position
+
+You can configure the `FlinkKinesisConsumer` to start reading the table-backing Kinesis data stream from a specific position through the `properties.flink.stream.initpos` option.
+Available values are:
+
+* `LATEST`: read all shards starting from the latest record.
+* `TRIM_HORIZON`: read all shards starting from the earliest record possible (data may be trimmed by Kinesis depending on the current retention settings of the backing stream).
+* `AT_TIMESTAMP`: read all shards starting from a specified timestamp. The timestamp must also be specified in the configuration
+properties by providing a value for `properties.flink.stream.initpos.timestamp`, in one of the following date patterns:
+   * A non-negative double value representing the number of seconds that has elapsed since the Unix epoch (for example, `1459799926.480`).
+   * A user defined pattern, which is a valid pattern for the `SimpleDateFormat` provided by `properties.flink.stream.initpos.timestamp.format`.
+     If a user-defined format is not supplied, the default pattern will be `yyyy-MM-dd'T'HH:mm:ss.SSSXXX`.
+     For example, timestamp value is `2016-04-04` and user-defined foramt is `yyyy-MM-dd`, or timestamp value is `2016-04-04T19:58:46.480-00:00` and a user-defined format is not provided.
+
+### Sink Partitioning
+
+Kinesis data streams consist of one or more shards, and the `sink.partitioner` option allows you to control how records written into a multi-shard Kinesis-backe table will be partitioned between its shards.
+Valid values are:
+
+* `fixed`: Kinesis `PartitionKey` values derived from the Flink subtask index, so each Flink partition ends up in at most one Kinesis partition (assuming that no re-sharding takes place at runtime).
+* `random`: Kinesis `PartitionKey`</code>` values are assigned randomly.
```

Comment:
```suggestion
* `random`: Kinesis `PartitionKey` values are assigned randomly.
```

Reply:
Done. Is there a syntax checker for the markdown files that I can run?

Reply:
I'm not aware of any. I usually just render the Flink docs and see if they are fine.